### PR TITLE
[RN] content component

### PR DESCRIPTION
**Pull request by brunokiafuka**
This PR contains the raw implementation of the `Content` component and its children in React Native.

> Some commented sections and HTML elements on the native components aren't currently being worked on as we are mainly focusing on stabilizing the `Editable` component with still needs to be worked on.

### CODE REVIEW

File: src/components/Alert.native.tsx [hunk 1]
```diff
@@ -5,7 +5,9 @@ import { TouchableOpacity, Text, StyleSheet } from 'react-native'
 import { Alert, State } from '../@types'
 import { View } from 'moti'
 import { commonStyles } from '../style/commonStyles'
+import { fadeIn } from '../style/animations'
 
+const { from, animate } = fadeIn
```
Comment (Foosballfan) [inline src/components/Alert.native.tsx:10]: tidy

File: src/components/Content.native.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+type ContentComponent = FC<ReturnType<typeof mapStateToProps>>
+
+/**
+ * Calculates whether there was a click on the left margin or padding zone of content element.
+ *
+ * @param e The onClick e object.
+ * @param content HTML element.
+ */
+const isLeftSpaceClick = (e: MouseEvent, content?: HTMLElement) => {
+  const style = window.getComputedStyle(content!)
```
Comment (Foosballfan) [inline src/components/Content.native.tsx:86]: you dont want this here

File: src/components/Content.native.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  /** Removes the cursor if the click goes all the way through to the content. Extends cursorBack with logic for closing modals. */
+  const clickOnEmptySpace = () => {
+    // make sure the the actual Content element has been clicked
+    // otherwise it will incorrectly be called on mobile due to touch vs click ordering (#1029)
+    if (!isPressed) return
+    setIsPressed(false)
+
+    // click event occured during text selection has focus node of type text unlike normal event which has node of type element
+    // prevent text selection from calling cursorBack incorrectly
+    const selection = window.getSelection()
```
Comment (Foosballfan) [inline src/components/Content.native.tsx:121]: again

File: src/components/ContentEditable.native.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  forceUpdate: boolean
+  onChange: (e: string) => void
+}
+
+/**
+ * Content Editable Component.
+ */
+const ContentEditable = ({ style, html, disabled, innerRef, forceUpdate, ...props }: ContentEditableProps) => {
+  const allowInnerHTMLChange = useRef<boolean>(true)
+  const contentRef = useRef<RichEditor>(null)
```
Comment (Foosballfan) [inline src/components/ContentEditable.native.tsx:21]: is this used since we ignore the toolbar?

File: src/components/ErrorMessage.native.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+const { flexEnd, whiteText, centerText, redBackground } = commonStyles
+const { from, animate } = fadeIn
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const mapStateToProps = ({ error }: State) => ({ value: error })
+
+/** An error message that can be dismissed with a close button. */
+const ErrorMessage = ({ value, dispatch }: Connected<{ value?: any }>) => (
+  <>
```
Comment (Foosballfan) [inline src/components/ErrorMessage.native.tsx:19]: unnecessary

File: src/components/NewThoughtInstructions.native.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  status,
+  tutorialStep,
+}) => {
+  const remoteLoading = status === 'connecting' || status === 'loading'
+
+  // loading
+  // show loading message if local store is loading or if remote is loading and there are no children
+  return (
+    <View /* className='new-thought-instructions' */>
+      {(localLoading || remoteLoading) && childrenLength === 0 ? (
```
Comment (Foosballfan) [inline src/components/NewThoughtInstructions.native.tsx:58]: these ternarys are real bad

File: src/components/Note.native.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    const noteThought = getParent(state, [...context, '=note'])
+    return !noteThought || noteThought.pending
+  }
+
+  if (!hasNote || isNotePending() || isContextViewActive(state, context)) return null
+
+  const note = attribute(state, context, '=note')
+
+  /** Handles note keyboard shortcuts. */
+  const onKeyDown = (e: React.KeyboardEvent) => {
```
Comment (Foosballfan) [inline src/components/Note.native.tsx:44]: seems irrelevant

File: src/components/Subthoughts.native.tsx [hunk 8]
```diff
@@ -0,0 +1,681 @@
...
+      )}
+    </>
+  )
+}
+
+SubthoughtsComponent.displayName = 'SubthoughtComponent'
+
+const Subthoughts = connect(mapStateToProps)(SubthoughtsComponent)
+
+export default Subthoughts
```
Comment (Foosballfan) [inline src/components/Subthoughts.native.tsx:681]: this file is crazy big

File: src/components/icons/UrlIcon.native.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import Svg, { Path } from 'react-native-svg'
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const UrlIcon = () => (
+  // eslint-disable-next-line react/jsx-no-undef
+  <Svg x='0px' y='0px' width='1em' height='1em' viewBox='0 0 30 30'>
```
Comment (Foosballfan) [inline src/components/icons/UrlIcon.native.tsx:7]: does RN allow em?

Comment (brunokiafuka) [reply src/components/icons/UrlIcon.native.tsx:7]: nah, just converted to dp

File: src/components/ContentEditable.native.tsx (hunk 4 above)
Comment (brunokiafuka) [reply src/components/ContentEditable.native.tsx:21]: not necessarily, this ref is to tell the toolbar which component is the editor component. To ignore the toolbar I had to set its height to 0

**Review by raineorshine — APPROVED** [review]

**Comment by raineorshine** [comment]
The CI failure is due to 33f658f9c5681894a673bb2bac9cce67d4651a76. There are no merge conflicts, but the `Path` type has been tightened up, which causes the type checker to fail. I can fix it once it is merged.
